logical/framework/field_type.go

```diff
@@ -13,6 +13,16 @@ const (
 	// TypeDurationSecond represent as seconds, this can be either an
 	// integer or go duration format string (e.g. 24h)
 	TypeDurationSecond
+
+	// TypeSlice represents a slice of any type
+	TypeSlice
+	// TypeStringSlice is a helper for TypeSlice that returns a sanitized
```

Comment:
I originally was going to clean trailing spaces and various other cleanup bits.  Do you think it makes sense to do any de-duping or cleaning of trailing spaces?